File: swift/Sources/allmark/RenderConsole/ConsoleEmphasisRenderer.swift
import Foundation

@MainActor
let consoleEmphasisRenderer = Renderer(
	name: "emphasis",
	render: renderConsoleEmphasis
)

@MainActor
func renderConsoleEmphasis(_ node: MarkdownNode, _ state: inout RendererState, _ first: Bool?, _ last: Bool?, _ decode: Bool?) {
	let styles = getConsoleStyles()
	let style = styles["emphasis"] ?? ""
	state.output += style
	renderChildren(node: node, state: &state)
	state.output += ansiReset
}

File: dotnet/Allmark/RenderConsole/ConsoleEmphasisRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class ConsoleEmphasisRenderer
{
    public static OutputRenderer Create()
    {
        return new OutputRenderer
        {
            Name = "emphasis",
            Render = Render,
        };
    }

    public static void Render(MarkdownNode node, RendererState state, bool? first = null, bool? last = null, bool? decode = true)
    {
        state.Output.Append($"{Ansi.Italic}{Ansi.Yellow}");
        RenderChildren.Execute(node, state);
        state.Output.Append($"{Ansi.Reset}");
    }
}
